### PR TITLE
Add material class

### PR DESCRIPTION
**Pull request by gabloe**

### CODE REVIEW

File: resources/shaders/shader.frag [hunk 1]
```diff
@@ -13,29 +13,36 @@ out vec4 color;
 // white light
 vec3 lightColor = vec3(1.0, 1.0, 1.0);
 
+struct Material {
+    vec4 ambient;
+    vec4 diffuse;
+    vec4 specular;
+    float shininess;
```
Comment (deathly809) [inline resources/shaders/shader.frag:20]: What is the difference between the two?

File: src/Resources/Material.cpp [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+  : _scene(scene), _mesh(mesh), _material{scene->mMaterials[mesh->mMaterialIndex]} {
+
+  Color = GetColor();
+  Shininess = GetShininess();
+  ShininessStrength = GetShininessStrength();
+}
+
+const float Material::GetShininessStrength() {
+  float shininess_strength = 1.0f;
+  _material->Get(AI_MATKEY_SHININESS_STRENGTH, shininess_strength);
```
Comment (deathly809) [inline src/Resources/Material.cpp:16]: Seems like something you should just cache.

File: src/Resources/Shader.cpp [hunk 3]
```diff
@@ -213,6 +213,26 @@ void ShaderProgram::setUniform(const std::string& name, int val) {
   glUniform1i(uniform(name), val);
 }
 
+void ShaderProgram::setVec3(const std::string& name, const glm::vec3& value) {
```
Comment (deathly809) [inline src/Resources/Shader.cpp]: These are just repeats of the above.

void ShaderProgram::setUniform(const std::string& name, const vec3& v) {
  glUniform3fv(uniform(name), 1, value_ptr(v));
}

File: resources/shaders/shader.frag (hunk 1 above)
Comment (deathly809) [reply resources/shaders/shader.frag:20]: shininess vs shininess_strength

Feels like if they want the strength large just make the shininess larger?

File: src/Includes/Shader.hpp [hunk 4]
```diff
@@ -80,6 +80,12 @@ class ShaderProgram {
   void setUniform(const std::string& name, float val);
   void setUniform(const std::string& name, int val);
```
Comment (deathly809) [inline src/Includes/Shader.hpp]: I just have setUniform(name, type) for all these above. That way I don't have to do setUniformType(name, type).

File: resources/shaders/shader.frag (hunk 1 above)
Comment (gabloe) [reply resources/shaders/shader.frag:20]: According to the docs:

SHININESS
Defines the shininess of a phong-shaded material. This is actually the exponent of the phong specular equation	
SHININESS=0 is equivalent to SHADING_MODEL=aiShadingMode_Gouraud.

SHININESS_STRENGTH	
Scales the specular color of the material.

**Pull request merged**